src/kafka/apache.rs

```diff
@@ -0,0 +1,321 @@
+///
+/// Module comes in two flavours:
+///
+/// - [`Apache Kafka GraalVM docker image`](https://hub.docker.com/r/apache/kafka-native), which is default as it provides faster startup and lower memory consumption.
+/// - [`Apache Kafka JVM docker image`](https://hub.docker.com/r/apache/kafka)
+///
+/// # Example
+/// ```
+/// use testcontainers_modules::{kafka::apache, testcontainers::runners::SyncRunner};
+/// let kafka_node = apache::Kafka::default().start().unwrap();
+///
+/// let bootstrap_servers = format!("127.0.0.1:{}", kafka_node.get_host_port_ipv4(apache::KAFKA_PORT).unwrap());
+///
+/// // connect to kafka server to send/receive messages
+/// ```
+
+#[derive(Debug, Clone)]
+pub struct Kafka {
+    env_vars: HashMap<String, String>,
+    image_name: String,
+    tag: String,
+}
+
+impl Default for Kafka {
+    fn default() -> Self {
+        let mut env_vars = HashMap::new();
+
+        env_vars.insert(
+            "KAFKA_LISTENERS".to_owned(),
+            format!(
+                "PLAINTEXT://0.0.0.0:{KAFKA_PORT},BROKER://0.0.0.0:9093,CONTROLLER://0.0.0.0:9094"
+            ),
+        );
+        env_vars.insert("CLUSTER_ID".to_owned(), DEFAULT_CLUSTER_ID.to_owned());
+        env_vars.insert(
+            "KAFKA_PROCESS_ROLES".to_owned(),
+            "broker,controller".to_owned(),
+        );
+
+        env_vars.insert(
+            "KAFKA_CONTROLLER_LISTENER_NAMES".to_owned(),
+            "CONTROLLER".to_owned(),
+        );
+        env_vars.insert(
+            "KAFKA_LISTENER_SECURITY_PROTOCOL_MAP".to_owned(),
+            "BROKER:PLAINTEXT,PLAINTEXT:PLAINTEXT,CONTROLLER:PLAINTEXT".to_owned(),
+        );
+        env_vars.insert(
+            "KAFKA_INTER_BROKER_LISTENER_NAME".to_owned(),
+            "BROKER".to_owned(),
+        );
+        env_vars.insert(
+            "KAFKA_ADVERTISED_LISTENERS".to_owned(),
+            format!("PLAINTEXT://localhost:{KAFKA_PORT},BROKER://localhost:9092",),
+        );
+        env_vars.insert("KAFKA_BROKER_ID".to_owned(), DEFAULT_BROKER_ID.to_string());
+        env_vars.insert(
+            "KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR".to_owned(),
+            DEFAULT_INTERNAL_TOPIC_RF.to_string(),
+        );
+        env_vars.insert(
+            "KAFKA_CONTROLLER_QUORUM_VOTERS".to_owned(),
+            format!("{DEFAULT_BROKER_ID}@localhost:9094").to_owned(),
+        );
+
+        Self {
+            env_vars,
+            image_name: KAFKA_NATIVE_IMAGE_NAME.to_string(),
+            tag: TAG.to_string(),
+        }
+    }
+}
+
+impl Kafka {
+    /// Switches default image to `apache/kafka` instead of `apache/kafka-native`
+    pub fn with_jvm_image(mut self) -> Self {
+        self.image_name = KAFKA_IMAGE_NAME.to_string();
+
+        self
+    }
+    /// Specify image tag
+    pub fn with_tag(mut self, tag: &str) -> Self {
+        self.tag = tag.to_string();
+        self
+    }
```

Comment:
Sorry, one more comment since it's been changed:  we don't usually expose methods that are unified across all images.

It's possible to use [ImageExt::with_tag](https://docs.rs/testcontainers/latest/testcontainers/core/trait.ImageExt.html#tymethod.with_tag) to override the tag

Reply:
all right, will revert commit

Reply:
done